Adding proper reference to stream for persistent subscription checkpoints (#4379)
EventStore/EventStore: docs/persistent-subscriptions.md

If a persistent subscription has a filter, then the persistent subscription will checkpoint when enough events are either handled or skipped by the filter.

> Checkpoint events have an Event Type of `$SubscriptionCheckpoint`, and are written to system streams that are named according to this format:

Reviewer: ```suggestion
Checkpoints are events written to system streams with event type `$SubscriptionCheckpoint`. Checkpoint stream name is composed using the subscribed stream name and consumer group name:
```

`$persistentsubscription-STREAMNAME::GROUPNAME-checkpoint`

> For example, if you have a Persistent Subscription group name of `LoanProcessing`, on top of the stream `$et-LoanRequested`, the system stream where checkpoints are written is called:

Reviewer: ```suggestion
For example, if you have a Persistent Subscription group name of `LoanProcessing`, on top of the stream `$et-LoanRequested`, the checkpoints stream name would be:
```
